src/topic_training/src/Begin_Here.md

```diff
@@ -0,0 +1,264 @@
+
+``` Python
+rospy.Subscriber('plot', Float64, callback)
+```
+Initialize the Subscriber. State the name of the topic (`plot`), the message type we will be publishing (float, which is actually in the class std_msgs.msg.Float64), and the function that will be called when a message is received (callBack(data)).
+
+``` Python
+pub = rospy.Publisher('chatter', String, queue_size=QUEUE_SIZE)
+```
+Initialize the Publisher. State the name of the topic (`chatter`), the message type we will be publishing (String, which is actually in the class std_msgs.msg.String), and the queue size (1). The queue size is the maximum number of messages kept as a buffer before throwing away old ones if we are publishing too quickly.
+
+The callback function for a Subscriber looks something like this:
+
+``` Python
+def callback(data: Float64) -> None:
+    global pi 
+    pi = data.data
+```
+This function will get called when a new message has arrived on the topic `plot`. The message received by the Subscriber, sent by the Publisher, is denoted as `msg`.
+
+Publishing a message on a topic would look something like this:
+
+``` Python
+pi_str = "currently at %s" % pi
+pub.publish(hello_str)
+```
+
+To break this down,
+
+``` Python
+pi_str = "currently at %s" % pi
+```
+Create the message.
+
+``` Python
+pub.publish(hello_str)
+```
+Publish the message using the previously initialized Publisher `pub`.
+
+If you have glanced at the C++ example code, you may have noticed the function `ros::spinOnce()` and its differences with `ros::spin()`. Python only has `rospy.spin()`, which can be completely replaced by a for loop, which is why you do not see it in the Python example code.
+
+## Running a Publisher and Subscriber
+Before you are able to run the example code, find the CMakeLists.txt file in the directory /workspaces/WRoverPlayground/src/topic_training. This file will include the the configurations, builds, and installs required to run your program as well as the files you will be running. Do not change anything in the file as it should already be set up for the demo files to work. 
+
+The `find_package`, `catkin_package`, and `include_directories` specifications are required for both the C++ and Python files to work. These provide the packages and directories needed for running ROS and ROS topics. 
+
+To run the C++ files, you will need the `add_executable`, `add_dependencies`, and `target_link_libraries` specifications. There will be three of each, representing the three C++ files in this section. `add_executable` determines which file you will use with the command `rosrun`, `add_dependencies` shows what the executable file depends on, and `target_link_libraries` links libraries or executable targets that may be used in the program. In larger projects, these specifications may contain different names, but since the code examples in this training are simple, they are all the same.
```

Comment:
I really like what you wrote here about what to include in the `CMakeLists`. It's important we have this documented somewhere as it's definitely hard for one to figure it out themselves without any outside help. But I feel like this should probably go elsewhere, maybe in another chapter or even an entirely new training module. It diverts focus away from the purpose of this training module (learning how to use chapters), and introduces a lot of new keywords that aren't needed for students to complete this training.

Reply:
I can see how new members could get overwhelmed quickly if they think we expect them to completely understand the `CMakeLists` file. What should I do with it? I also wrote something for the `rqt_*` commands that Ben told me to remove and save since it should be placed somewhere else. What training should these two explanations be?

Reply:
That's a great question, thanks for bringing this up.

@bennowotny I'm curious what's your opinion on this? The `rqt` related writeup can go in the Debug training, but what about `CMakeLists`? I feel like this could either be it's own training or as a standalone documentation somewhere in the repo.

Reply:
The CMake Training should move to Setup. @co-li

Reply:
Fixed in a2e2adb.